tools/earth/landcover_subindic.xml

```
@@ -0,0 +1,110 @@
+            --raster_1 '$raster_1'
+            --raster_2 '$raster_2'
+            --json '$json'
+    ]]></command>
+    <inputs>
+        <param name="raster_1" type="data" format="tiff" label="A raster image of the land at a reference year" />
+        <param name="raster_2" type="data" format="tiff" label="A raster image of the land at the year you want to study" />
+        <param name="json" type="data" format="json" label="A file containing the land specifications" help="You can directly use the default one here https://github.com/fair-ease/earth-critical-zone-galaxy/blob/main/landcover.json but you can also customize it"/>
+    </inputs>
+    <outputs>
+        <collection name="output_tiff" type="list" label="Land cover raster outputs">
+            <discover_datasets pattern="__name_and_ext__" directory="data/land_cover/output/" />
+        </collection>
+        <!--<collection name="output_png" type="list" label="Land cover plots">
+            <discover_datasets pattern="__name_and_ext__" directory="data/land_cover/" />
+        </collection>-->
+        <data name="contour" format="shp"  from_work_dir="data/land_cover/output/change_yf_yi0.shp" label="${tool.name} on ${on_string}" />
+    </outputs>
+    <tests>
+        <test expect_num_outputs="2">
+            <param name="raster_1" value="EU_CLC_1990.tiff"/>
+            <param name="raster_2" value="EU_CLC_2018.tiff"/>   
+            <param name="json" value="landcover.json"/>
+            <output name="contour" value="change_yf_yi0.shp"/>
+            <output_collection name="output_tiff" type="list" count="7">
+                <element name="lc_baseline_esa" ftype="tiff">
+                    <assert_contents>
+                        <has_text text="LAEA Europe"/>
+                    </assert_contents>
+                </element>
+                <element name="lc_bl" ftype="tiff">
+                    <assert_contents>
+                        <has_text text="LAEA Europe"/>
```

wow, does this work? How can a tiff file has text?

---

hum it works yes usually I let it fail on purpose in order to check what will work. So this was part of it. But no idea how ....